### PR TITLE
Heedls 594 deactive admins test fix

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-594

### Description
Added logic to prevent accessing the endpoints when they should not be able to be accessed e.g. for your own account.
Added logic to delete admin user records when they have no logins.
Updated action link text on confirmation page.

### Screenshots
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [x] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [x] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs [hunk 1]
```diff
@@ -156,12 +163,19 @@ public IActionResult DeactivateAdmin(int adminId, int? returnPage)
         [ServiceFilter(typeof(VerifyAdminUserCanAccessAdminUser))]
         public IActionResult DeactivateAdmin(int adminId, DeactivateAdminViewModel model)
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs:164]: As deletion can be a potentially serious thing, would it be worth renaming this method (and the get) to DeactivateOrDeleteAdmin just for developer's benefit/safety?

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Overall, pretty good. Few minor quibbles.

File: DigitalLearningSolutions.Data.Tests/DataServices/SessionDataServiceTests.cs [hunk 2]
```diff
@@ -4,16 +4,15 @@
...
     using DigitalLearningSolutions.Data.Tests.TestHelpers;
     using FluentAssertions;
     using NUnit.Framework;
 
     internal class SessionDataServiceTests
     {
-        private const int twoMinutesInMilliseconds = 120 * 1000;
-        private SessionDataService sessionDataService;
-        private SessionTestHelper sessionTestHelper;
+        private const int TwoMinutesInMilliseconds = 120 * 1000;
```
Comment (SteveJacksonSoft) [inline DigitalLearningSolutions.Data.Tests/DataServices/SessionDataServiceTests.cs:13]: Private consts should be camel case I think.

File: DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs [hunk 3]
```diff
@@ -169,46 +179,110 @@ public void UnlockAccount_unlocks_account_and_returns_to_page()
...
+
+        [Test]
+        public void DeactivateAdminUser_returns_not_found_when_trying_to_access_page_for_own_admin_account()
+        {
+            // Given
+            var adminUser = UserTestHelper.GetDefaultAdminUser();
+            var loggedInAdminUser = adminUser;
+
+            A.CallTo(() => userDataService.GetAdminUserById(adminUser.Id)).Returns(adminUser);
+            A.CallTo(() => userDataService.GetAdminUserById(loggedInAdminUser.Id)).Returns(loggedInAdminUser);
```
Comment (JBsoftwire) [inline DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs:197]: Doesn't this just duplicate the previous line?

**Review by JBsoftwire — CHANGES_REQUESTED** [review]
I have a nitpick, but I may just have misunderstood something.

File: DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs [hunk 4]
```diff
@@ -144,26 +145,39 @@ public IActionResult UnlockAccount(int adminId)
         [Route("{adminId:int}/DeactivateAdmin")]
         [HttpGet]
         [ServiceFilter(typeof(VerifyAdminUserCanAccessAdminUser))]
-        public IActionResult DeactivateAdmin(int adminId, int? returnPage)
+        public IActionResult DeactivateOrDeleteAdmin(int adminId, int? returnPage)
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Controllers/TrackingSystem/Centre/Administrator/AdministratorController.cs:148]: Looks like the action name specified in the form has not been updated to match this, so we start getting problems when trying to submit it.

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Couple of quick changes still necessary, but looking pretty good otherwise.

File: DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs (hunk 3 above)
Comment (DanBloxham-sw) [reply DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs]: Different IDs return different users. Needed for getting past the checks to ensure a user does not deactivate themselves.

Comment (DanBloxham-sw) [reply DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Centre/Administrator/AdministratorControllerTests.cs]: Ah right. Think I was getting confused with a different test. Looks good to me now.

**Review by DanBloxham-sw — APPROVED** [review]
This is looking good now. I'll assign the ticket back to you, up to you if you want to wait for the others to approve it.

**Pull request merged**